### PR TITLE
Add a root collections endpoint.

### PR DESCRIPTION
**Pull request by DailyDreaming**
Addresses #1932.

This endpoint adds the `GET /collections` root endpoint which allows a user to retrieve all of their collections without knowing the UUID for each one.

For example:
```
res = self.app.get('/v1/collections',
                    headers=get_auth_header(authorized=True),
                    params=dict(replica='aws'))
print(res.json())
```
Should return something like this:
```
{
  "collections": [
    {
      "collection": {
        "contents": [
          {
            "type": "file",
            "uuid": "29aad27a-536a-42c3-92df-e7c5a14df22d",
            "version": "2019-04-02T165421.995240Z"
          },
          {
            "fragment": "/foo",
            "type": "foo",
            "uuid": "29aad27a-536a-42c3-92df-e7c5a14df22d",
            "version": "2019-04-02T165421.995240Z"
          }
        ],
        "description": "d",
        "details": {},
        "name": "n",
        "owner": "service-account@b.gserviceaccount.com"
      },
      "collection_uuid": "69649f49-595e-4d88-897b-430a904ae491",
      "collection_version": "2019-04-02T095424.559988Z"
    },
    {
      "collection": {
        "contents": [
          {
            "type": "file",
            "uuid": "29aad27a-536a-42c3-92df-e7c5a14df22d",
            "version": "2019-04-02T165421.995240Z"
          },
          {
            "fragment": "/foo",
            "type": "foo",
            "uuid": "29aad27a-536a-42c3-92df-e7c5a14df22d",
            "version": "2019-04-02T165421.995240Z"
          }
        ],
        "description": "d",
        "details": {},
        "name": "n",
        "owner": "service-account@b.gserviceaccount.com"
      },
      "collection_uuid": "8fedec52-b528-4d88-9d1b-83dab039ea69",
      "collection_version": "2019-04-02T095423.287298Z"
    }
  ]
}
```
A follow-up PR will be made to add paging to this once this is merged.

### CODE REVIEW

File: dss/api/collections.py [hunk 1]
```diff
@@ -46,6 +46,24 @@ def get_impl(uuid: str, replica: str, version: str = None):
         raise DSSException(404, "not_found", "Could not find collection for UUID {}".format(uuid))
     return json.loads(collection_blob)
 
+@dss_handler
+@security.authorized_group_required(['hca'])
+def listc(replica: str):
```
Comment (xbrianh) [inline dss/api/collections.py]: Due to the manner in which collections are assembled in the DSS, this list is likely to be long. We should consider adding support for paging relatively soon.

File: dss-api.yml [hunk 2]
```diff
@@ -996,23 +996,75 @@ paths:
                 required:
                   - code
   /collections:
+    get:
+      operationId: dss.api.collections.listc
```
Comment (amarjandu) [inline dss-api.yml]: is this supposed to be `dss.api.collections.listcollection` ?

File: dss/api/collections.py (hunk 1 above)
Comment (DailyDreaming) [reply dss/api/collections.py]: I completely agree.  Should have a PR up pretty soon after this is merged.

File: dss-api.yml (hunk 2 above)
Comment (DailyDreaming) [reply dss-api.yml]: No, but I can change it!

Comment (DailyDreaming) [reply dss-api.yml]: Addressed.

**Pull request merged**